app/src/main/java/com/example/drawingactivity/ForcableEditText.kt

```diff
@@ -0,0 +1,52 @@
+package com.example.drawingactivity
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.text.TextWatcher
+import android.util.AttributeSet
+import android.widget.EditText
+
+@SuppressLint("ClickableViewAccessibility")
+class ForcableEditText @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : EditText(context, attrs, defStyle) {
+
+    override fun addTextChangedListener(watcher: TextWatcher?) {
+        watcher?.let { if (!textListeners.contains(it)) textListeners.add(it) }
+        super.addTextChangedListener(watcher)
+    }
+
+    private val textListeners = mutableListOf<TextWatcher>()
+
+    private fun removeAllListeners() {
+        textListeners.forEach { removeTextChangedListener(it) }
+    }
+
+    private fun addBackAllListeners() {
+        textListeners.forEach { addTextChangedListener(it) }
+    }
+
+    fun forceText(text: String) {
+        if (!this.hasFocus()) {
+            removeAllListeners()
+            setText(text)
+            addBackAllListeners()
+        }
+    }
+
+    /**
+     * Checks to confirm if the string input is an int or not.
+     * If it is an int, converts it to an int.
+     */
+    val checkIfInt: Int?
+        get() {
+            return this.text.toString().toIntOrNull()
+        }
+
+    val floatValue: Float
```

Comment:
What if this fails? Why not make this nullable like the other? `floatValueOrNull`?

Reply:
so I started to introduce this and it added a lot of complexity through null checks when I already am null checking before doing the calculations with an if statement. Since I generally just need to check if the value in the input can change to a full integer (I am not allowing decimal value entries in any of the text fields).

If you feel strongly that there is a high chance that users could get around these null checks and still do the math, I could implement all of this.